code_search/code-search.ipynb

```diff
@@ -273,13 +279,78 @@
     "gbq.read_gbq(query, dialect='standard', project_id=PROJECT)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define an experiment\n",
+    "\n",
+    "* This solution consists of multiple jobs and servers that need to share parameters\n",
+    "* To facilitate this we use [experiments.libsonnet](https://github.com/kubeflow/examples/blob/master/code_search/kubeflow/components/experiments.libsonnet) to define sets of parameters\n",
+    "* Each set of parameters has a name corresponding to a key in the dictionary defined in **experiments.libsonnet**\n",
+    "* You configure an experiment by defining a set of experiments in **experiments.libsonnet**\n",
+    "* You then set the global parameter **experiment** to the name of the experiment defining the parameters you want\n",
+    "  to use.\n",
+    "\n",
+    "To get started define your experiment\n",
+    "\n",
+    "* Create a new entry containing a set of values to be used for your experiment\n",
+    "* Pick a suitable name for your experiment.\n",
+    "* Set the following values\n",
+    "\n",
+    "    * **outputDir**: The GCS directory where the output should be written\n",
+    "    * **train_steps**: Numer oftraining steps\n",
+    "    * **eval_steps**: Number of steps to be used for eval\n",
+    "    * **hparams_set**: The set of hyperparameters to use; see some suggestions [here](https://github.com/tensorflow/tensor2tensor#language-modeling).\n",
+    "    \n",
+    "        * **transformer_tiny** can be used to train a very small model suitable for ensuring the code works.\n",
+    "    * **project**: Set this to the GCP project you can use\n",
+    "    * **modelDir**: \n",
+    "        * After training your model set this to a GCS directory containing the export model\n",
+    "        * e.g gs://code-search-demo/models/20181107-dist-sync-gpu/export/1541712907/\n",
+    "    * **problem**: set this to \"kf_github_function_docstring\",\n",
+    "    * **model**: set this  \"kf_similarity_transformer\",\n",
+    "    * **lookupFile**: set this to the GCS location of the CSV produced by the job to create the nmslib index of the embeddings for all GitHub data\n",
+    "    * **indexFile*:* set this to the GCS location of the nmslib index for all the data in GitHub\n",
+    "\n",
+    "Configure your ksonnet environment to use your experiment\n",
+    "\n",
+    "* Open **kubeflow/environments/${ENVIRONMENT}/globals.libsonnet**\n",
+    "* Define the following global parameters\n",
+    "\n",
+    "  * **experiment**: Name of your experiment; should correspond to a key defined in **experiments.libsonnet**\n",
+    "  * **project**: Set this to the GCP project you can use\n",
```

Comment:
I am wondering why not move all param to experiments.libsonnet, except experiment name ?
the project argument for example is duplicated in two places.

Reply:
I think that's where we're headed: kubeflow/examples#308
This PR was already quite big so I didn't want to try to clean everything up in this PR.

I'm not sure whether it makes sense to move everything into experiments or define some other things as globals.